pysingfel/detector/jungfrau.py

```diff
@@ -1,17 +1,124 @@
 import re
 import six
+import numpy as np
 
 from pysingfel.util import deprecated
 from .lcls import LCLSDetector
 
-class JungfrauDetector(LCLSDetector):
-    def __init__(self, *args, **kwargs):
-        super(JungfrauDetector, self).__init__(*args, **kwargs)
+class JungfrauDetector(AutoRangingDetector):
+    def __init__(self, cameraConfig, *args, **kwargs):
+        super(JungfrauDetector, self).__init__(cameraConfig=cameraConfig, *args, **kwargs)
 
+    cameraConfigs = [None, 
+                         "highLow", "mediumLow", ## autoranging ePix10k
+                         "fixedHigh", "fixedMedium", "fixedLow"] ## fixed
+
+        self.nBits = 14
+        self.maxEnergy = 10000*8 ## keV
+        self.nRanges = 2
+        if self.cameraConfig in ["mediumLow", "fixedMedium", "fixedLow"]:
+            self.switchPoints = [300*8, self.maxEnergy*666] ## keV
+            if self.cameraConfig == "fixedMedium":
+                tmp = self.switchPoints[0]
+                self.switchPoints[0] = self.maxEnergy*666
+                self.maxEnergy = tmp
+            if self.cameraConfig == "fixedLow":
+                self.switchPoints[0] = -666
+            self.switchPointVariations = [4.*8, 0.] ## keV, placeholder
+            self.switchPointWidths = [4.*8, 0.] ## keV, placeholder
+            self.offsets = [0, 2100] ## ADU; placeholder
+            self.gains = [4.1, 0.1] ## ADU/keV
+            self.gainErrors = [0.01, 0.015] ## dimensionless, placeholder
+            self.nonLinearity = [0.01/2400, 0.017/self.maxEnergy] ## 1/keV, placeholder
+        else:
+            self.switchPoints = [100*8, self.maxEnergy*666] ## keV
+            if self.cameraConfig == "fixedHigh":
+                tmp = self.switchPoints[0]
+                self.switchPoints[0] = self.maxEnergy*666
+                self.maxEnergy = tmp
+            self.switchPointVariations = [2.*8, 0.] ## keV, placeholder
+            self.switchPointWidths = [2.*8, 0.] ## keV, placeholder
+            self.offsets = [0, 1100] ## ADU; placeholder
+            self.gains = [12.5, 0.1] ## ADU/keV
+            self.gainErrors = [0.01, 0.015] ## dimensionless, placeholder
+            self.nonLinearity = [0.01/800, 0.017/self.maxEnergy] ## 1/keV, placeholder
```

Comment:
Most of this looks identical to Epix10k case. If it's the same for all AutoRangingDetectors, then that part should be set in the AutoRangingDector's constructor.

Reply:
fixed